src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -373,43 +374,40 @@ void SpecCache::Populate(
       const int lowerBoundX = jj == 0 ? 0 : copyEnd;
       const int upperBoundX = jj == 0 ? copyBegin : numPixels;
 
-// todo(mhodgkinson): I don't find an option to define _OPENMP anywhere. Is this
-// still of interest?
-#ifdef _OPENMP
-      // Storage for mutable per-thread data.
-      // private clause ensures one copy per thread
-      struct ThreadLocalStorage {
-         ThreadLocalStorage()  { }
-         ~ThreadLocalStorage() { }
-
-         void init(SampleTrackCache &waveTrackCache, size_t scratchSize) {
-            if (!cache) {
-               cache = std::make_unique<SampleTrackCache>(waveTrackCache.GetTrack());
-               scratch.resize(scratchSize);
-            }
-         }
-         std::unique_ptr<SampleTrackCache> cache;
-         std::vector<float> scratch;
-      } tls;
-
-      #pragma omp parallel for private(tls)
-#endif
-      std::optional<AudioSegmentSampleView> sampleCacheHolder;
-      for (auto xx = lowerBoundX; xx < upperBoundX; ++xx)
-      {
-#ifdef _OPENMP
-         tls.init(waveTrackCache, scratchSize);
-         SampleTrackCache& cache = *tls.cache;
-         float* buffer = &tls.scratch[0];
-#else
-         float* buffer = &scratch[0];
-#endif
-         CalculateOneSpectrum(sampleCacheHolder,
-            settings, clip, xx, pixelsPerSecond, lowerBoundX, upperBoundX,
-            gainFactors, buffer, &freq[0]);
+      if (!reassignment) {
+         auto first = NumberIterator<>{ lowerBoundX },
+            last = NumberIterator<>{ upperBoundX };
+         const auto calcOne = [&,
+            sampleCacheHolder = std::optional<AudioSegmentSampleView>{},
+            scratch = std::vector<float>(scratchSize)
+         ](int xx) mutable {
+            float* buffer = &scratch[0];
+            CalculateOneSpectrum(sampleCacheHolder,
+               settings, clip, xx, pixelsPerSecond, lowerBoundX, upperBoundX,
+               gainFactors, buffer, &freq[0]);
+         };
+         using namespace Parallel;
+         ForEach task{ first, last, calcOne };
+         OneDimensionalReduce(task);
```

Comment:
If I understand correctly, this function may put CPU to 100% load during the playback. There must be more control over the thread priorities and count here.